cuebot/src/main/java/com/imageworks/spcue/servant/ManageJob.java

```diff
@@ -417,9 +429,15 @@ public void setMinGpus(JobSetMinGpusRequest request, StreamObserver<JobSetMinGpu
     public void setPriority(JobSetPriorityRequest request, StreamObserver<JobSetPriorityResponse> responseObserver) {
         try{
             setupJobData(request.getJob());
-            jobDao.updatePriority(job, request.getVal());
-            responseObserver.onNext(JobSetPriorityResponse.newBuilder().build());
-            responseObserver.onCompleted();
+            if (isJobFinished()) {
+                responseObserver.onError(Status.FAILED_PRECONDITION
```

Comment:
Currently this block that populates the error response gets repeated a lot. Is there a way to write a function that applies the check and populates the error response in one go, to decrease the repetition?

Reply:
Regarding repetition, I am not exactly sure how to solve this problem. Do you happen to have any pointers?

Reply:
Added a suggestion bellow